File: app/src/main/java/com/monster/app/niceweather/activities/main/mvp/view/MainView.java
package com.monster.app.niceweather.activities.main.mvp.view;

import android.view.View;

import com.monster.app.niceweather.models.ForecastCityModel;

import java.util.List;

import io.reactivex.Observable;

/**
 * Created by monster on 2/25/18.
 */

public interface MainView {

    View getView();

    Observable<Object> observeRefresh();

    Observable<Integer> observeItemClick();

    void setLoading(boolean loading);

    void setForecastCityItems(List<ForecastCityModel> items);

}

File: app/src/main/java/com/monster/app/niceweather/activities/main/mvp/view/MainView.kt
package com.monster.app.niceweather.activities.main.mvp.view

import android.view.View

import com.monster.app.niceweather.models.ForecastCityModel

import io.reactivex.Observable

/**
 * Created by monster on 2/25/18.
 */

interface MainView {

    val view: View

    val observeRefresh: Observable<Any>

    val observeItemClick: Observable<ForecastCityModel>

    fun setLoading(loading: Boolean)

    fun setForecastCityItems(items: List<ForecastCityModel>)

    fun showError(message: String)

}
